spec.html

```diff
@@ -32497,173 +32497,107 @@ <h1>String.prototype.replace ( _searchValue_, _replaceValue_ )</h1>
             1. Let _stringLength_ be the number of code units in _str_.
             1. Assert: _position_ &le; _stringLength_.
-            1. Assert: _captures_ is a possibly empty List of Strings.
-            1. Assert: Type(_replacement_) is String.
+            1. Assert: _captures_ is a possibly empty List, each of whose elements is a String or *undefined*.
+            1. Assert: Type(_replacer_) is String.
+            1. Let _preceding_ be the substring of _str_ from 0 to _position_.
             1. Let _tailPos_ be _position_ + _matchLength_.
-            1. Let _m_ be the number of elements in _captures_.
-            1. Let _result_ be the String value derived from _replacement_ by copying code unit elements from _replacement_ to _result_ while performing replacements as specified in <emu-xref href="#table-replacement-text-symbol-substitutions"></emu-xref>. These `$` replacements are done left-to-right, and, once such a replacement is performed, the new replacement text is not subject to further replacements.
-            1. Return _result_.
+            1. If _tailPos_ &ge; _stringLength_, then
+              1. Let _following_ be the empty String.
+            1. Else,
+              1. Let _following_ be the substring of _str_ from _tailPos_.
+            1. Let _interpretNumericRef_ be a new Abstract Closure with parameters (_ref_) that captures _captures_ and performs the following steps when called:
+              1. Assert: _ref_ starts with *"$"*.
+              1. Let _digits_ be the substring of _ref_ from 1.
+              1. Assert: _digits_ consists of 1 or 2 decimal digits.
+              1. Let _index_ be the integer that results from interpreting _digits_ as a decimal numeral.
+              1. Assert: 0 &le; _index_ &le; 99.
+              1. If _index_ = 0 or _index_ &gt; the number of elements in _captures_, then
+                1. NOTE: No replacement is done.
+                1. Return _ref_.
+              1. Else,
+                1. Let _capture_ be the _index_<sup>th</sup> element of _captures_.
+                1. If _capture_ is *undefined*, then
+                  1. Return the empty String.
+                1. Else,
+                  1. Return _capture_.
+            1. Let _replacerRemainder_ be _replacer_.
+            1. Let _replacement_ be the empty String.
+            1. Repeat, while _replacerRemainder_ is not the empty String,
+              1. If _replacerRemainder_ starts with *"$$"*, then
+                1. Let _ref_ be *"$$"*.
+                1. Let _refReplacement_ be *"$"*.
+              1. Else if _replacerRemainder_ starts with *"$&amp;"*, then
+                1. Let _ref_ be *"$&amp;"*.
+                1. Let _refReplacement_ be _matched_.
+              1. Else if _replacerRemainder_ starts with *"$`"*, then
+                1. Let _ref_ be *"$`"*.
+                1. Let _refReplacement_ be _preceding_.
+              1. Else if _replacerRemainder_ starts with *"$'"*, then
```

Comment:
The kerning here is... not great: <img width="34" alt="Screen Shot 2021-08-06 at 11 58 38 PM" src="https://user-images.githubusercontent.com/1653598/128591599-2b9adf76-ff9d-4763-ba64-12711ce25e3e.png">

Not sure if there's an easy fix, though.

Reply:
Agreed, not great. The status quo doesn't have this problem because it uses backtick markdown, but this isn't really what we use backticks for now.

We could say `... starts with [a string consisting of] the code unit 0x0024 (DOLLAR SIGN) followed by the code unit 0x0027 (APOSTROPHE)`, but that's not great either.

Reply:
(no change here yet)